### PR TITLE
[develop2] reference for conan install cmd

### PR DESCRIPTION
**Pull request by memsharded**

### CODE REVIEW

File: reference/commands/install.rst [hunk 1]
```diff
@@ -1,2 +1,273 @@
...
+- Download precompiled binaries, or build binaries from sources in the local cache, in the right order for the dependency graph.
+- Create the necessary files as requested by the "generators", so build systems and other tools can locate the locally installed dependencies
+- Optionally, execute desired ``deployers``.
+
+
+Conanfile path or --requires
+----------------------------
+
+The ``conan install`` command can use 2 different origins for information. The first one is using a local ``conanfile.py`` 
+or ``conanfile.txt``, containing definitions of dependencies, generators to be used.
```
Comment (RubenRBS) [inline reference/commands/install.rst:115]:
```suggestion
or ``conanfile.txt``, containing definitions of the dependencies and generators to be used.
```

File: reference/commands/install.rst [hunk 2]
```diff
@@ -1,2 +1,273 @@
...
+    $ conan install --requires=zlib/1.2.13 --tool-requires=ninja/1.11.0
+
+
+In the general case, it is recommended to use a ``conanfile`` instead of defining things in the command line.
+
+
+Profiles, Settings, Options, Conf
+---------------------------------
+
+There are several arguments that are used to define the effective profiles that will be used, both for "build"
```
Comment (RubenRBS) [inline reference/commands/install.rst:150]:
```suggestion
There are several arguments that are used to define the effective profiles that will be used, both for the "build"
```

File: reference/commands/install.rst [hunk 3]
```diff
@@ -1,2 +1,273 @@
...
+By default the arguments refer to the "host" context, so ``--settings:host, -s:h`` is totally equivalent to
+``--settings, -s``. Also, by default, the ``conan install`` command will use the ``default`` profile both for the
+"build" and "host" context. That means that if a profile with the "default" name has not been created, it will error.
+
+Multiple definitions of profiles can be passed as arguments, and they will compound from left to right (right has the
+highest priority)
+
+.. code-block:: bash
+
+    # The values of myprofile 3 will have higher priority
```
Comment (RubenRBS) [inline reference/commands/install.rst:162]:
```suggestion
    # The values of myprofile3 will have higher priority
```

File: reference/commands/install.rst [hunk 4]
```diff
@@ -1,2 +1,273 @@
...
+    $ conan install . -pr=myprofile1 -pr=myprofile2 -pr=myprofile3
+
+If values of ``settings``, ``options`` and ``conf`` are provided in the command line, they create a profile that
```
Comment (RubenRBS) [inline reference/commands/install.rst:165]:
```suggestion
If values for any of ``settings``, ``options`` and ``conf`` are provided in the command line, they create a profile that
```

File: reference/commands/install.rst [hunk 5]
```diff
@@ -1,2 +1,273 @@
...
+is composed with the other provided ``-pr`` (or the "default" one if not specified) profiles, with higher priority,
```
Comment (RubenRBS) [inline reference/commands/install.rst:166]:
```suggestion
is composed with the other provided ``-pr`` profiles (or the "default" one if not specified), which has the highest priority,
```

File: reference/commands/install.rst [hunk 6]
```diff
@@ -1,2 +1,273 @@
...
+not matter what is the order of arguments).
```
Comment (RubenRBS) [inline reference/commands/install.rst:167]:
```suggestion
not matter what the order of arguments is.
```

File: reference/commands/install.rst [hunk 7]
```diff
@@ -1,2 +1,273 @@
...
+
+    # the final "host" profile will always be build_type=Debug, even if "myprofile"
+    # says "build_type=Release"
+    $ conan install . -pr=myprofile -s build_type=Debug
+    
+
+Generators and deployers
+------------------------
+
+The ``-g`` argument allows to define in command line the different built-in generators to be used:
```
Comment (RubenRBS) [inline reference/commands/install.rst:179]:
```suggestion
The ``-g`` argument allows to define in the command line the different built-in generators to be used:
```

File: reference/commands/install.rst [hunk 8]
```diff
@@ -1,2 +1,273 @@
...
+
+.. code-block:: bash
+
+    $ conan install --requires=zlib/1.2.13 -g CMakeDeps -g CMakeToolchain
+
+Note that in the general case, the recommended approach is to have the ``generators`` defined in the ``conanfile``, 
+and only for the ``--requires`` use case, it would be more necessary as command line argument.
+
+Generators are intended to create files for the build systems to locate the dependencies, while the ``deployers``
+main use case is copy files from the Conan cache to user space, and other custom operations over the dependency graph,
```
Comment (RubenRBS) [inline reference/commands/install.rst:189]:
```suggestion
main use case is to copy files from the Conan cache to user space, and performing any other custom operations over the dependency graph,
```

File: reference/commands/install.rst [hunk 9]
```diff
@@ -1,2 +1,273 @@
...
+  structure to avoid conflicts between files and artifacts of different packages
+- ``direct_deploy`` does a copy of only the immediate direct dependencies, but does not include the transitive
+  dependencies.
+
+
+Some generators might have the capability of redefining the target "package folder". That means that if some other
+generator like ``CMakeDeps`` is used that is pointing to the packages, it will be pointing to the local deployed
+copy, and not to the original packages in the Conan cache.
+
+It is also possible, and it is a powerful extension point, to write user custom deployers.
```
Comment (RubenRBS) [inline reference/commands/install.rst:210]:
```suggestion
It is also possible, and it is a powerful extension point, to write custom user deployers.
```

File: reference/commands/install.rst [hunk 10]
```diff
@@ -1,2 +1,273 @@
...
+Read more about custom deployers in :ref:`reference_extensions_deployers`.
+
+
+Name, version, user, channel
+----------------------------
+
+The ``conan install`` command provides optional arguments for ``--name, --version, --user, --channel``. These 
+arguments might not be necessary in the majority of cases. Never for ``conanfile.txt`` and for ``conanfile.py``
```
Comment (RubenRBS) [inline reference/commands/install.rst:218]: What does it mean here:
> Never for ``conanfile.txt`` and for ``conanfile.py``

That `--name` etc should not be used if using a `conanfile...`? Or am I misunderstanding the meaning?

File: reference/commands/install.rst [hunk 11]
```diff
@@ -1,2 +1,273 @@
...
+
+    # If we don't specify ``--version``, it will be None and it will fail
+    $ conan install . --version=3.24
+
+
+Lockfiles
+---------
+
+The ``conan install`` command has several arguments to load and produce lockfiles. 
+By default, if a ``conan.lock`` file is located besides the recipe or in the current working directory
```
Comment (RubenRBS) [inline reference/commands/install.rst:245]:
```suggestion
By default, if a ``conan.lock`` file is located beside the recipe or in the current working directory
```

File: reference/commands/install.rst [hunk 12]
```diff
@@ -1,2 +1,273 @@
...
+if no path is provided, will be used as an input lockfile. 
+
+Lockfiles are strict by default, that means that
+if there is some ``requires`` and it cannot find a matching locked reference in the lockfile, it will error
+and stop. For cases where it is expected that the lockfile will not be complete, as there might be new
+dependencies, the ``--lockfile-partial`` can be used.
```
Comment (RubenRBS) [inline reference/commands/install.rst:251]:
```suggestion
dependencies, the ``--lockfile-partial`` argument can be used.
```

File: reference/commands/install.rst [hunk 13]
```diff
@@ -1,2 +1,273 @@
...
+
+By default, ``conan install`` will not generate an output lockfile, but if the ``--lockfile-out`` argument
+is provided, pointing to a filename, like ``--lockfile-out=result.lock``, then a lockfile will be generated
+from the current dependency graph. If ``--lockfile-clean`` argument is provided, all versions and revisions
+not used in the current dependency graph will be dropped from the resulting lockfile.
+
+Let`s say that we already have a ``conan.lock`` input lockfile, but we just added a new ``requires = "newpkg/1.0"``
```
Comment (RubenRBS) [inline reference/commands/install.rst:258]:
```suggestion
Let's say that already we have a ``conan.lock`` input lockfile, but we just added a new ``requires = "newpkg/1.0"``
```

File: reference/commands/install.rst [hunk 14]
```diff
@@ -1,2 +1,273 @@
...
+to a new dependency. We could resolve the dependencies, locking all the previously locked versions, while allowing
+to resolve the new one, not previously in the lockfile, and store it in a new or overwrite the existing lockfile:
```
Comment (RubenRBS) [inline reference/commands/install.rst:260]:
```suggestion
to resolve the new one, which was not previously present in the lockfile, and store it in a new location, or overwrite the existing lockfile:
```

File: reference/commands/install.rst [hunk 15]
```diff
@@ -1,2 +1,273 @@
...
+
+.. code-block:: bash
+
+    # --locfile=conan.lock is the default, not necessary
+    $ conan install . --lockfile=conan.lock --lockfile-partial --lockfile-out=conan.lock 
+
+The ``--lockfile-packages`` allows to create lockfiles that also contain down to the package revision, but
```
Comment (RubenRBS) [inline reference/commands/install.rst:267]:
```suggestion
The ``--lockfile-packages`` argument allows to create lockfiles that also lock down to the package revision, but 
```

**Review by RubenRBS — APPROVED** [review]
I hope that you don't hate me after all these nitpicking suggestions!

File: reference/commands/install.rst (hunk 13 above)
Comment (memsharded) [reply reference/commands/install.rst:258]: I think this one was good?

File: reference/commands/install.rst (hunk 5 above)
Comment (memsharded) [reply reference/commands/install.rst:166]: Maybe this one is not clear, it is composed, but not clear if the which applies to the "default" profile (it doesn't, it applies to the settings and options one)

File: reference/commands/install.rst (hunk 10 above)
Comment (memsharded) [reply reference/commands/install.rst:218]: Never for ``conanfile.txt``.
It only makes sense for ``conanfile.py``

**Comment by memsharded** [comment]
> I hope that you don't hate me after all these nitpicking suggestions!

This review was fantastic, thanks very much for doing it!

File: reference/commands/install.rst (hunk 13 above)
Comment (RubenRBS) [reply reference/commands/install.rst:258]: oops, yeah it's only the ` vs '

**Review by czoido — APPROVED** [review]

**Pull request merged**